allennlp/training/optimizers.py

```diff
@@ -0,0 +1,38 @@
+from typing import Any, Dict, List, Union
+
+import torch
+
+from ..common.checks import ConfigurationError
+
+optimizers = {  # pylint: disable=invalid-name
+        "adam": torch.optim.Adam,
+        "adagrad": torch.optim.Adagrad,
+        "adadelta": torch.optim.Adadelta,
+        "sgd": torch.optim.SGD,
+        "rmsprop": torch.optim.RMSprop,
+
+}
+
+
+def get_optimizer_from_params(model_parameters: List[torch.Tensor],
+                              params: Union[Dict[str, Any], str]) -> torch.optim.Optimizer:
+    """
+    This function converts a set of model parameters to be optimized and a
+    dictionary of optimizer parameters (or a string name of one of the optimizers above)
+    into a Pytorch optimizer object which can apply gradient updates to the model
+    parameters following model.backward() calls.
+    The simplest case for both of these is a string that shows up in `optimizers`
```

Comment:
You need double back ticks for code (throughout this docstring), single back tick is italics.